codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddAwsAuthPlugin.java

```diff
@@ -44,6 +45,7 @@
 /**
  * Configure clients with AWS auth configurations and plugin.
  */
+// TODO: Think about AWS Auth supported for only some operations and not all, when not AWS service, with say @auth([])
```

Comment:
If the service has an auth trait you need to add the mechanics needed to support that regardless of what the default is (where `@auth([])` is no auth by default). Since the `auth` trait can only reference auth traits on the service, you don't have to worry about a single operation introducing its own auth. I'm pretty sure the current setup already handles operations with disabled auth.

Reply:
I put this TODO to think more deeply about different cases when I handle supporting SigV4 for non-AWS clients.